### PR TITLE
Add console-telemetry-plugin 

### PR DESCRIPTION
**Pull request by divyanshiGupta**
**Fixes:** https://issues.redhat.com/browse/ODC-6522

This PR adds console-telemetry-plugin to console packages with telemetry-listener for capturing events in segment.

https://user-images.githubusercontent.com/20724543/169876469-394ace3a-5927-4e3d-b62a-3af91963cbfb.mov

### CODE REVIEW

File: frontend/packages/console-telemetry-plugin/src/flags/detect-telemetry.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { SetFeatureFlag } from '@console/dynamic-plugin-sdk';
+
+export const detectTelemetry = (setFeatureFlag: SetFeatureFlag) => {
+  const { disabled: isTelemetryDisabled = false } = window.SERVER_FLAGS.telemetry;
+  setFeatureFlag('TELEMETRY', !isTelemetryDisabled);
```
Comment (jerolimov) [inline frontend/packages/console-telemetry-plugin/src/flags/detect-telemetry.ts:5]: `window.SERVER_FLAGS.telemetry` flags should be defined as `Record<string, string>`. We should define a type for this somewhere.

```suggestion
  const { disabled } = window.SERVER_FLAGS.telemetry;
  setFeatureFlag('TELEMETRY', disabled !== 'true');
```

File: frontend/packages/console-telemetry-plugin/src/telemetry.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      analytics._loadOptions = e;
+    };
+    analytics.SNIPPET_VERSION = '4.13.1';
+    if (segmentKey) {
+      analytics.load(segmentKey);
+    }
+  }
+};
+
+initSegment();
```
Comment (jerolimov) [inline frontend/packages/console-telemetry-plugin/src/telemetry.ts:70]: Let us initialise segment only if the segment key is defined (and disabled it is not `'true'`).

File: frontend/packages/console-telemetry-plugin/src/telemetry.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
+/* eslint-disable no-console */
```
Comment (jerolimov) [inline frontend/packages/console-telemetry-plugin/src/telemetry.ts:1]: can you move and rename this file to src/listeners/segment.ts as this is a specific implementation? Wdyt?

File: frontend/packages/console-telemetry-plugin/console-extensions.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    "properties": {
+      "handler": { "$codeRef": "telemetryFlags.detectTelemetry" }
+    }
+  },
+
+  {
+    "type": "console.telemetry/listener",
+    "properties": {
+      "listener": {
+        "$codeRef": "telemetry.eventListener"
```
Comment (jerolimov) [inline frontend/packages/console-telemetry-plugin/console-extensions.json:13]: I would recommend making it clear that this is an specific implementation

```suggestion
        "$codeRef": "telemetry.listeners.segment"
```

File: frontend/packages/console-telemetry-plugin/src/listeners/segment.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+SEGMENT_API_KEY && !TELEMETRY_DISABLED && initSegment();
+
+const anonymousIP = {
+  context: {
+    ip: '0.0.0.0',
+  },
+};
+
+export const eventListener = async (eventType: string, properties?: any) => {
```
Comment (jerolimov) [inline frontend/packages/console-telemetry-plugin/src/listeners/segment.ts:75]: Should we use the `TelemetryEventListener` here? What happen if the Promise fail? Will this crash our console? (Not tested by myself.) In this case we should maybe wrap this code in a try-catch?

```suggestion
export const eventListener: TelemetryEventListener = async (eventType: string, properties?: any) => {
```

File: frontend/packages/console-telemetry-plugin/src/listeners/segment.ts [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+      if (n.parentNode) {
+        n.parentNode.insertBefore(t, n);
+      }
+      // eslint-disable-next-line no-underscore-dangle
+      analytics._loadOptions = e;
+    };
+    analytics.SNIPPET_VERSION = '4.13.1';
+    if (SEGMENT_API_KEY) {
+      analytics.load(SEGMENT_API_KEY);
+    }
```
Comment (jerolimov) [inline frontend/packages/console-telemetry-plugin/src/listeners/segment.ts:63]: We call this method only when API key is defined, right?

```suggestion
    analytics.load(SEGMENT_API_KEY);
```

File: frontend/packages/console-telemetry-plugin/src/listeners/segment.ts (hunk 5 above)
Comment (divyanshiGupta) [reply frontend/packages/console-telemetry-plugin/src/listeners/segment.ts]: This extension is used in `useTelemetry` hook which gets it using `useResolvedExtensions` hook which handles this case.

**Review by jerolimov — APPROVED** [review]
Looks good to me, but want check this one more time a cluster bot instance. :+1:

/approve

**Comment by jerolimov** [comment]
Epic has no UX, doesn't need docs and I tested this on a cluster bot instance:

/label px-approved
/label qe-approved
/label docs-approved
/lgtm

**Pull request merged**